samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -250,6 +343,413 @@ def test_post_created(self):
+        self.assertEqual(self.get_tl_event_count('create'), 0)
+        self.assertEqual(self.get_app_alert_count('create'), 0)
+        response = self.request_knox(
+            self.url, 'POST', data=self.post_data, token=self.token_admin
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        ticket = IrodsAccessTicket.objects.first()
+        self.assertEqual(ticket.path, self.path)
+        self.assertEqual(ticket.label, self.label)
+        self.assertEqual(
+            ticket.date_expires,
+            datetime.strptime(self.date_expires, '%Y-%m-%dT%H:%M:%S.%f%z'),
+        )
+        self.assertEqual(ticket.user, self.user)
+        self.assertEqual(ticket.study, self.study)
+        self.assertEqual(ticket.assay, self.assay)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 1)
+        self.assertEqual(self.get_app_alert_count('create'), 2)
+        self.assertEqual(
+            self.app_alert_model.objects.filter(
+                alert_name='irods_ticket_create'
+            )
+            .first()
+            .user,
+            self.user_delegate,
+        )
+
+        # Assert ticket state in iRODS
+        irods_ticket = self.get_irods_ticket(ticket)
+        self.assertEqual(irods_ticket[TicketQuery.Ticket.type], 'read')
+        self.assertIsNotNone(irods_ticket[TicketQuery.Ticket.expiry_ts])
+        self.assertEqual(
+            irods_ticket[TicketQuery.Collection.name], self.coll.path
+        )
+
+    def test_create_contributor(self):
+        """Test POST IrodsAccessTicketCreateAPIView as contributor"""
+        self.assertEqual(IrodsAccessTicket.objects.count(), 0)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 0)
+        self.assertEqual(self.get_app_alert_count('create'), 0)
+        response = self.request_knox(
+            self.url, 'POST', data=self.post_data, token=self.token_contrib
+        )
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        ticket = IrodsAccessTicket.objects.first()
+        self.assertEqual(ticket.user, self.user_contrib)
+        self.assert_alert_count(CREATE_ALERT, self.user, 0)
+        self.assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self.assertEqual(self.get_tl_event_count('create'), 1)
+        self.assertEqual(self.get_app_alert_count('create'), 3)
+        self.assertEqual(
+            self.app_alert_model.objects.filter(
+                alert_name='irods_ticket_create'
+            )
+            .first()
+            .user,
+            self.user_delegate,
+        )
+
+    def test_create_no_expiry(self):
```

Comment:
### Note:
I've tried to replace the `with self.login()` block with 
```
response = self.request_knox(
            self.url, 'POST', data=self.post_data, token=self.token_admin
        )
```
But for some reason it started failing (
Idk why, but I let it as is

Reply:
What is `self.token_admin`? If you are doing this as superuser, the default token is used if you simply leave the `token` argument out.

Also, what exactly is the error? We should consistently use `request_knox()` in testing these views and it should work with every view.

Reply:
`self.token_admin` is a token for superuser. I thought, that the default token is for contributor.. 
`request_knox()` didn't work due to datetime format. However, changing it for None value solved the problem

Reply:
Yeah, sure:
Just change `self.post_data['date_expires'] =  None` to   `self.post_data['date_expires'] =  ''`.
Then the test will fail:
```
======================================================================
FAIL: test_create_no_expiry (samplesheets.tests.test_views_api_taskflow.TestIrodsAccessTicketCreateAPIView)
Test POST IrodsAccessTicketCreateAPIView with no expiry date
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gromdimon/Working/sodar-server/samplesheets/tests/test_views_api_taskflow.py", line 442, in test_create_no_expiry
    self.assertEqual(response.status_code, 201)
AssertionError: 400 != 201

----------------------------------------------------------------------

```

The content of reponse is: `'{"date_expires":["Datetime has wrong format. Use one of these formats instead: YYYY-MM-DDThh:mm[:ss[.uuuuuu]][+HH:MM|-HH:MM|Z]."]}'`